src/libcore/sync/atomic.rs

```diff
@@ -941,8 +942,10 @@ impl<T> AtomicPtr<T> {
     ///
     /// let other_ptr   = &mut 10;
     /// let another_ptr = &mut 10;
-    ///
-    /// let value = some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed);
+    /// // here `some_ptr` will not change as `other_ptr` has a different address from `ptr`
+    /// asserteq!(some_ptr.compare_and_swap(other_ptr, another_ptr, Ordering::Relaxed), 5);
```

Comment:
Also, `some_ptr` *will* change here (returning the previous value of 5), because both arguments passed to the `compare_and_swap` are equal.